public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  dateMath,
+  DateTime,
+  MutableDataFrame,
+  DataSourceApi,
+  DataSourceInstanceSettings,
+  DataQueryRequest,
+  DataQueryResponse,
+  DataQuery,
+  FieldType,
+} from '@grafana/data';
+import { getBackendSrv, BackendSrvRequest } from '@grafana/runtime';
+import { Observable, from, of } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { getTimeSrv, TimeSrv } from 'app/features/dashboard/services/TimeSrv';
+import { serializeParams } from 'app/core/utils/fetch';
+
+export type TempoQuery = {
+  query: string;
+} & DataQuery;
+
+export class TempoDatasource extends DataSourceApi<TempoQuery> {
+  constructor(private instanceSettings: DataSourceInstanceSettings, private readonly timeSrv: TimeSrv = getTimeSrv()) {
+    super(instanceSettings);
+  }
+
+  async metadataRequest(url: string, params?: Record<string, any>): Promise<any> {
+    const res = await this._request(url, params, { hideFromInspector: true }).toPromise();
+    return res.data.data;
+  }
+
+  query(options: DataQueryRequest<TempoQuery>): Observable<DataQueryResponse> {
+    // At this moment we expect only one target. In case we somehow change the UI to be able to show multiple
+    // traces at one we need to change this.
+    const id = options.targets[0]?.query;
+    if (id) {
+      return this._request(`/api/traces/${encodeURIComponent(id)}`).pipe(
+        map(response => {
+          return {
+            data: [
+              new MutableDataFrame({
+                fields: [
+                  {
+                    name: 'trace',
+                    type: FieldType.trace,
+                    values: response?.data?.data || [],
+                  },
+                ],
+                meta: {
+                  preferredVisualisationType: 'trace',
+                },
+              }),
+            ],
+          };
+        })
+      );
+    } else {
+      return of({
+        data: [
+          new MutableDataFrame({
+            fields: [
+              {
+                name: 'trace',
+                type: FieldType.trace,
+                values: [],
+              },
+            ],
+            meta: {
+              preferredVisualisationType: 'trace',
+            },
+          }),
+        ],
+      });
+    }
+  }
+
+  async testDatasource(): Promise<any> {
+    return true;
```

Comment:
shouldn't this try to do a real request?

Reply:
yeah forgot about it in jaeger, added a simple test there